src/lib/components/UrlChooserContainer/index.js

```diff
@@ -0,0 +1,78 @@
+import {html} from "lit-element";
+import {store} from "../../store";
+import {BaseElement} from "../BaseElement";
+import {userRef, firebase} from "../../fb";
+import "../UrlChooser";
+
+/**
+ * @fileoverview Manages state interaction with UrlChooser.
+ *
+ * Invokes Lighthouse when the UrlChooser requests it, possibly with an updated URL.
+ */
+
+/* eslint-disable require-jsdoc */
+class UrlChooserContainer extends BaseElement {
+  static get properties() {
+    return {
+      url: {type: String},
+      active: {type: Boolean},
+    };
+  }
+
+  constructor() {
+    super();
+    this.url = null; // when signed out or waiting for Firestore, this is null
+    this.active = false;
+
+    store.subscribe(this.onStateChanged.bind(this));
+    this.onStateChanged();
+  }
+
+  render() {
+    return html`
+      <web-url-chooser
+        .url=${this.url}
+        .active=${this.active}
+        @audit=${this.runAudit}
+      ></web-url-chooser>
+    `;
+  }
+
+  onStateChanged() {
+    const state = store.getState();
+
+    // if Lighthouse is active, ignore changes to the URL from Firestore
```

Comment:
is this checking if lighthouse is running or not? Maybe we should have explicit state flags for `isLighthouseAuditing` or something along those lines? Currently I'm having trouble telling at a glance what the relationship between `activeLighthouseUrl` and `active` is.